stdlib/public/core/ContiguousArrayBuffer.swift

```diff
@@ -718,21 +718,48 @@ internal struct _ContiguousArrayBuffer<Element>: _ArrayBufferProtocol {
     _internalInvariant(bounds.upperBound <= count)
 
     let initializedCount = bounds.upperBound - bounds.lowerBound
-    target.initialize(
-      from: firstElementAddress + bounds.lowerBound, count: initializedCount)
+    
+    //Future: we could relax the count requirement, but we'd need to handle
+    //incompletely consuming the buffer rather than just setting it to the
+    //empty buffer singleton
+    if self.count == initializedCount && isUniquelyReferenced() {
+      _ = beginCOWMutation()
```

Comment:
Doing `beginCOWMutation` immediately after `isUniquelyReferenced` feels super sketchy to me, gotta say

Reply:
You can directly use `beginCOWMutation`:

```
  if !_isPOD(Element.self) && self.count == initializedCount && beginCOWMutation() {
```